src/factories/nodes.ts

```diff
@@ -45,7 +46,12 @@ export async function nodesContainerFactory(runId: string) {
 
   container.name = DEFAULT_NODES_CONTAINER_NAME
 
-  emitter.on('layoutUpdated', () => renderNodes())
+  emitter.on('layoutUpdated', () => {
+    nodesLayout = null
+    rows.clear()
+    columns.clear()
+    renderNodes()
+  })
```

Comment:
This also fixed some issues where row offsets from one view would carry over when the view was switched. Seems like a good idea anyway to clear things out when the view is changed.